doc/development/static_code_analysis.md

```diff
@@ -0,0 +1,22 @@
+# Static Code Analysis
+
+This Open Source project uses [SonarCloud](https://sonarcloud.io/) as static code analysis tool.
+
+## Rules and Exceptions
+
+We are using Sonar's **standard rule set** with the following exceptions-
+
+### Code Duplication Check Exceptions
+
+The SQL dialects and the metadata readers that are part of them are in many parts quite similar. Most duplications are one-liners that configure the behavior of a SQL dialect. This can't be reduced any further without sacrificing self-documentation of dialects and proper abstraction.
```

Comment:
```suggestion
The SQL dialects and the metadata readers that are part of dialects are in many places quite similar. Most duplications are one-liners that configure the behavior of a SQL dialect. This can't be reduced any further without sacrificing self-documentation of dialects and proper abstraction.
```